rct229/rulesets/ashrae9012019/section5/section5rule41.py

```diff
@@ -0,0 +1,89 @@
+    get_opaque_surface_type,
+    OpaqueSurfaceType as OST,
+)
+
+ABSORPTION_THERMAL_EXTERIOR = 0.9
+PASS_NOT_EQUAL_MSG = (
+    "Roof surface emittance in P-RMD matches that in U-RMD but is not equal to 0.9"
+)
+PASS_DIFFERS_MSG = "Roof thermal emittance is equal to the prescribed default value of 0.9 but differs from the thermal emittance in the user model"
+
+
+class Section5Rule41(RuleDefinitionListIndexedBase):
+    """Rule 41 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule41, self).__init__(
+            rmrs_used=UserBaselineProposedVals(True, False, True),
+            each_rule=Section5Rule41.BuildingRule(),
+            index_rmr="proposed",
+            id="5-41",
+            description="The proposed roof surfaces shall be modeled using the same thermal emittance as in the user model.",
+            ruleset_section_title="Envelope",
+            standard_section="Section G3.1-1(a) Building Envelope Modeling Requirements for the Proposed design",
+            is_primary_rule=True,
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule41.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(True, False, True),
+                each_rule=Section5Rule41.BuildingRule.RoofRule(),
+                index_rmr="baseline",
+                list_path="$.building_segments[*].zones[*].surfaces[*]",
+            )
+
+        def list_filter(self, context_item, data=None):
+            surface_p = context_item.proposed
+            return get_opaque_surface_type(surface_p) == OST.ROOF
+
+        class RoofRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section5Rule41.BuildingRule.RoofRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(True, False, True),
+                    required_fields={
+                        "$": ["surface_optical_properties"],
+                        "surface_optical_properties": ["absorptance_thermal_exterior"],
+                    },
+                )
+
+            def get_calc_vals(self, context, data=None):
+                roof_p = context.proposed
+                roof_u = context.user
+
+                return {
+                    "absorptance_thermal_exterior_p": roof_p[
+                        "surface_optical_properties"
+                    ]["absorptance_thermal_exterior"],
+                    "absorptance_thermal_exterior_u": roof_u[
+                        "surface_optical_properties"
+                    ]["absorptance_thermal_exterior"],
+                }
+
+            def rule_check(self, context, calc_vals=None, data=None):
+                return (
+                    calc_vals["absorptance_thermal_exterior_p"]
+                    == calc_vals["absorptance_thermal_exterior_u"]
```

Comment:
@weilixu This looks like it is giving PASS outcome for when absorptance_thermal_exterior_p == absorptance_thermal_exterior_u but absorptance_thermal_exterior_p != ABSORPTION_THERMAL_EXTERIOR. This is not the desired outcome. 

Based on Maria's comments on the RDS, we can simplify the PASS condition to be when absorptance_thermal_exterior_p == ABSORPTION_THERMAL_EXTERIOR. 

Only the PASS message will depend on if absorptance_thermal_exterior_u is also equal.